Assets/VRTK/Scripts/Interactions/VRTK_InteractGrab.cs

```diff
@@ -316,17 +336,32 @@ protected virtual void SetControllerAttachPoint()
             if (controllerReference.model != null && originalControllerAttachPoint == null)
             {
                 //attempt to find the attach point on the controller
-                Transform defaultAttachPoint = controllerReference.model.transform.Find(VRTK_SDK_Bridge.GetControllerElementPath(SDK_BaseController.ControllerElements.AttachPoint, VRTK_DeviceFinder.GetControllerHand(interactTouch.gameObject)));
-                if (defaultAttachPoint != null)
-                {
-                    controllerAttachPoint = defaultAttachPoint.GetComponent<Rigidbody>();
+                SDK_BaseController.ControllerHand handType = VRTK_DeviceFinder.GetControllerHand(interactTouch.gameObject);
+                string elementPath = VRTK_SDK_Bridge.GetControllerElementPath(SDK_BaseController.ControllerElements.AttachPoint, handType);
+                attemptSetCurrentControllerAttachPoint = StartCoroutine(SetCurrentControllerAttachPoint(elementPath, 10, 0.1f));
+            }
+        }
 
-                    if (controllerAttachPoint == null)
-                    {
-                        Rigidbody autoGenRB = defaultAttachPoint.gameObject.AddComponent<Rigidbody>();
-                        autoGenRB.isKinematic = true;
-                        controllerAttachPoint = autoGenRB;
-                    }
+        protected virtual IEnumerator SetCurrentControllerAttachPoint(string searchPath, int attempts, float delay)
+        {
+            Transform defaultAttachPoint = controllerReference.model.transform.Find(searchPath);
+
+            while (defaultAttachPoint == null && attempts > 0)
+            {
+                defaultAttachPoint = controllerReference.model.transform.Find(searchPath);
+                attempts--;
+                yield return new WaitForSeconds(delay);
```

Comment:
what was your suggestion for this again about sticking something something in variables or something?